### PR TITLE
set logger as static final

### PR DESCRIPTION
**Pull request by Nowheresly**
use of StringBuilder instead of StringBuffer
call dispose() on graphics to avoid memory leak
use logger instead of calling e.printStackTrace()

thanks for taking time to review

### CODE REVIEW

File: src/main/java/com/kodcu/service/MarkdownService.java [hunk 1]
```diff
@@ -21,7 +21,7 @@
 @Component
 public class MarkdownService {
 
-    private final Logger logger = LoggerFactory.getLogger(MarkdownService.class);
+    private static final Logger logger = LoggerFactory.getLogger(MarkdownService.class);
```
Comment (lefou) [inline src/main/java/com/kodcu/service/MarkdownService.java]: What's the motivation of making it static here?

`MarkdownService` is supposed to be used as singleton, so no space savings. OTOH you risk potentially threading issues depending on the Logger implementation.

**Comment by lefou** [comment]
I'd prefer to not use static for loggers, if there is no obvious gain.

**Comment by rahmanusta** [comment]
Oh, if it is OK for you @Nowheresly , can you rool them back ?

**Comment by Nowheresly** [comment]
@lefou thanks for your feedback 
Well using static logger or not is a matter of taste.
From the official documentation 
http://www.slf4j.org/faq.html#declared_static
'We no longer recommend one approach over the other '
Previously there were some logger declared as static, some other not.
We should at least be consistent and document it.

**Pull request merged**